learn/ai_powered_search/rest_response_request.mdx

```diff
@@ -0,0 +1,254 @@
+
+Most providers accept either a string or an array of strings. A single string will generate one request per document in your database. An array of strings allows Meilisearch to send up to 10 documents in one request.
+
+Update your embedder object specifying your chosen input format. If you were using Mistral, your request would look like this:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"]
+    }
+  }
+}
+```
+
+Use `{{text}}` to indicate Meilisearch should replace the contents of a field with your document data. If using an array, like in this example, also specify `{{..}}`. This instructs Meilisearch to include multiple documents in a single request.
+
+When using other tools, `input` might be called something else, like `text`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "request": {
+      "text": "{{text}}"
+    }
+  }
+}
+```
+
+## Other request fields
+
+You may add as many fields to the `request` object as you need. Meilisearch will send them as configured when it queries the embeddings provider.
+
+For example, Mistral allows you to optionally configure an `encoding_format`. Set it by declaring this field in your embedder's `request`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"],
+      "encoding_format": "float"
+    }
+  }
+}
+```
+
+## The embedding response
+
+Each provider responds to queries using its own specific structure. You must indicate where Meilisearch can find the document embeddings in the provider's response.
+
+Consult your provider's API documentation, paying attention to where it places the embeddings. Then describe the full path to the embedding array in your embedder's `response`.
+
+Mistral's embeddings are included in `response.data`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "request": {
+      "model": "MODEL_NAME",
+      "input": ["{{text}}", "{{..}}"],
+      "encoding_format": "float"
+    },
+    "response": {
+      "data": [
+        {
+          "embedding": "{{embedding}}"
+        },
+        "{{..}}"
+      ]
+    }
+  }
+}
+```
+
+Embeddings are located in the `embedding` field of each item in an array of objects. The first item in `response.data` tells Meilisearch where to find the embedding, while the second item, `"{{..}}"`, indicates there are multiple embeddings in the response.
+
+Cloudflare's embeddings are instead located in `response.result.data`:
+
+```json
+{
+  "EMBEDDER_NAME": {
+    "source": "rest",
+    "request": {
+      "text": "{{text}}"
+    },
+    "response": {
+      "result": {
+        "data": "{{embedding}}"
```

Comment:
Cloudflare seems to always return the data as an array of embeddings, even when `text` is a single string and not an array.

So you should have 
```suggestion
        "data": "[{{embedding}}]"
```